File: src/source_location.rs
use derive_more::Display;

#[derive(Debug, Clone, PartialEq, Display)]
#[display(fmt = "{}:{}:{}", filepath, line, column)]
pub struct SourceLocation {
    pub filepath: String,
    pub position: usize,
    pub line: usize,
    pub column: usize,
}

File: Langite/Syntax/SourceLocation.cs
namespace Langite.Syntax
{
    public readonly struct SourceLocation
    {
        public SourceLocation(string filepath, int position, int line, int column)
        {
            Filepath = filepath;
            Position = position;
            Line = line;
            Column = column;
        }

        public string Filepath { get; }
        public int Position { get; }
        public int Line { get; }
        public int Column { get; }

        public override string ToString()
        {
            return $"{Filepath}:{Line}:{Column}";
        }
    }
}
